Sdk/Speckle.Connectors.Utils/Instances/InstanceObjectsManager.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Speckle.Core.Models.Instances;
+
+namespace Speckle.Connectors.Utils.Instances;
+
+/// <summary>
+/// Abstract class to cover common functionalities of inheritors.
+/// </summary>
+public abstract class InstanceObjectsManager<THostObjectType, TAppIdMapValueType>
```

Comment:
make this a separate class and not a base class.  Share logic with injected dependencies instead of through inheritance.  Makes things cleaner and more testable in the long run